deeposlandia/datasets/dataset.py

```diff
@@ -0,0 +1,161 @@
+"""Encapsulates datasets in a generic Dataset class, and in some more specific
+classes that inherit from it
+
+"""
+
+import os
+import json
+import math
+from multiprocessing import Pool
+
+import cv2
+from PIL import Image
+
+import numpy as np
+
+from deeposlandia import utils
+
+class Dataset:
```

Comment:
If the method `populate` must be implemented, the code should raise an error if it's not the case. Maybe we should use the `abstractmethod`  decorator to force this.

This should be the case for all methods or properties we must have in order to have a well-defined subclass.

Reply:
I create one single abstract method, namely `populate`, by using `@abc.abstractmethod` decorator.